src/runtime_src/core/tools/xbmgmt2/OO_Hotplug.cpp

```diff
@@ -0,0 +1,198 @@
+#include "core/common/system.h"
+
+// 3rd Party Library - Include Files
+#include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
+#include <boost/algorithm/string.hpp>
+namespace po = boost::program_options;
+
+// System - Include Files
+#include <fstream>
+// =============================================================================
+
+// ----- H E L P E R M E T H O D S ------------------------------------------
+static void
+hotplug_online()
+{
+  static const std::string  rescan_path = "/sys/bus/pci/rescan";
+
+  if (!boost::filesystem::exists(rescan_path))
+    throw xrt_core::error((boost::format("Invalid sysfs file path '%s'.") % rescan_path).str());
+
+  std::ofstream rescan_file(rescan_path, std::ofstream::out);
+  try
+  {
+    if (!rescan_file.is_open())
+      throw xrt_core::error((boost::format("Unable to open the sysfs file '%s'.") % rescan_path).str());
+
+    // Writing "1" to /sys/bus/pci/rescan will trigger the hotplug event.
+    rescan_file << 1;
+    rescan_file.flush();
+    if (!rescan_file.good()) {
+      rescan_file.close();
+      throw std::runtime_error(boost::str(boost::format("Can't write to file %s") % rescan_path));
+    }
+  }
+  catch(const xrt_core::error& e) {
+    std::cerr << boost::format("\nERROR: %s\n") % e.what();
+    throw xrt_core::error(std::errc::operation_canceled);
+  }
+
+  rescan_file.close();
+}
+
+// ----- C L A S S   M E T H O D S -------------------------------------------
+
+OO_Hotplug::OO_Hotplug(const std::string &_longName, bool _isHidden )
+    : OptionOptions(_longName, _isHidden, "Perform hotplug for the given device")
+{
+  m_optionsDescription.add_options()
+    ("device,d", po::value<decltype(m_devices)>(&m_devices), "The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest")
+    ("action", po::value<decltype(m_action)>(&m_action)->required(), "Action to perform: online or offline")
+    ("help", po::bool_switch(&m_help), "Help to use this sub-command")
+  ;
+
+  m_positionalOptions.
+    add("action", 1 /* max_count */)
+  ;
+}
+
+void
+OO_Hotplug::execute(const SubCmdOptions& _options) const
+{
+  XBUtilities::verbose("SubCommand option: Hotplug");
+
+  XBUtilities::verbose("Option(s):");
+  for (const auto & aString : _options)
+    XBUtilities::verbose(std::string(" ") + aString);
+
+  // Honor help option first
+  if (std::find(_options.begin(), _options.end(), "--help") != _options.end()) {
+    printHelp();
+    return;
+  }
+
+  // Parse sub-command ...
+  po::variables_map vm;
+
+  try {
+    po::store(po::command_line_parser(_options).options(m_optionsDescription).positional(m_positionalOptions).run(), vm);
+    po::notify(vm); // Can throw
+  }
+  catch (po::error& e) {
+    std::cerr << "ERROR: " << e.what() << "\n\n";
+    printHelp();
+    throw xrt_core::error(std::errc::operation_canceled);
+  }
+
+  if (m_help) {
+    printHelp();
+    return;
+  }
+
+  // Exit if neither action or device specified
+  if (m_action.empty()) {
```

Comment:
Hmm...If action is required to be spectified with a value, does that make this conditional statement obsolete for boost::program_options is peforming this check.

Reply:
Removed. Thanks